### PR TITLE
.NET 5.0 RC2 API  updates

### PR DESCRIPTION
**Pull request by gewarren**

### CODE REVIEW

**Comment by BillWagner** [comment]
Any idea why this would have merge conflicts? That seems odd for the build pipeline.

**Comment by gewarren** [comment]
> Any idea why this would have merge conflicts? That seems odd for the build pipeline.

Thanks, I didn't see that but it's resolved now. I had to add that method back manually due to https://ceapex.visualstudio.com/Engineering/_workitems/edit/298401.

**Review by BillWagner — APPROVED** [review]
I skimmed a set of the changes, and this LGTM.

You can :shipit: when ready.

File: xml/System.Formats.Cbor/CborWriter.xml [hunk 1]
```diff
@@ -0,0 +1,1312 @@
...
+      </Attributes>
+      <ReturnValue>
+        <ReturnType>System.Void</ReturnType>
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="value" Type="System.UInt32" />
+      </Parameters>
+      <Docs>
+        <param name="value">The value to write</param>
+        <summary>Writes anvalue as an unsigned integer encoding (major type 0).</summary>
```
Comment (gewarren) [inline xml/System.Formats.Cbor/CborWriter.xml:1192]:
```suggestion
        <summary>Writes a value as an unsigned integer encoding (major type 0).</summary>
```

File: xml/System.Formats.Cbor/CborWriter.xml [hunk 2]
```diff
@@ -0,0 +1,1312 @@
...
+      </Attributes>
+      <ReturnValue>
+        <ReturnType>System.Void</ReturnType>
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="value" Type="System.UInt64" />
+      </Parameters>
+      <Docs>
+        <param name="value">The value to write</param>
+        <summary>Writes an value as an unsigned integer encoding (major type 0).</summary>
```
Comment (gewarren) [inline xml/System.Formats.Cbor/CborWriter.xml:1231]:
```suggestion
        <summary>Writes a value as an unsigned integer encoding (major type 0).</summary>
```

File: xml/System.Formats.Cbor/CborWriter.xml [hunk 3]
```diff
@@ -0,0 +1,1312 @@
...
+      </AssemblyInfo>
+      <ReturnValue>
+        <ReturnType>System.Void</ReturnType>
+      </ReturnValue>
+      <Parameters>
+        <Parameter Name="value" Type="System.Int32" />
+      </Parameters>
+      <Docs>
+        <param name="value">The value to write</param>
+        <summary>Writes an value as a signed integer encoding (major types 0,1)</summary>
```
Comment (gewarren) [inline xml/System.Formats.Cbor/CborWriter.xml:740]:
```suggestion
        <summary>Writes a value as a signed integer encoding (major types 0,1)</summary>
```

File: xml/System.Globalization/CompareInfo.xml [hunk 4]
```diff
@@ -2222,13 +2222,23 @@ Use <xref:System.Globalization.CompareInfo.GetSortKeyLength(System.ReadOnlySpan{
...
+        <param name="options">The <see cref="T:System.Globalization.CompareOptions" /> to use during the search.</param>
+        <param name="matchLength">When this method returns, contains the number of characters of <paramref name="source" /> that matched the desired value. This may be different than the length of <paramref name="value" /> if a linguistic comparison is performed. Set to 0 if <paramref name="value" /> is not found within <paramref name="source" />.</param>
+        <summary>Searches for the first occurrence of a substring within a source string.</summary>
+        <returns>The zero-based index into <paramref name="source" /> where the substring <paramref name="value" /> first appears; or a negative value if <paramref name="value" /> cannot be found within <paramref name="source" />.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This method has greater overhead than other <xref:System.Globalization.CompareInfo.IndexOf%2A> overloads which don't take a `matchLength` argument. Call this overload only if you require the match length information.
```
Comment (gewarren) [inline xml/System.Globalization/CompareInfo.xml:2236]:
```suggestion
This method has greater overhead than other <xref:System.Globalization.CompareInfo.IndexOf%2A> overloads that don't take a `matchLength` argument. Call this overload only if you require the match length information.
```

File: xml/System.Globalization/CompareInfo.xml [hunk 5]
```diff
@@ -3092,18 +3102,29 @@ Use <xref:System.Globalization.CompareInfo.GetSortKeyLength(System.ReadOnlySpan{
...
+        <param name="matchLength">When this method returns, contains the number of characters of <paramref name="source" /> that matched the desired prefix. This may be different than the length of <paramref name="prefix" /> if a linguistic comparison is performed. Set to 0 if the prefix did not match.</param>
+        <summary>Determines whether a string starts with a specific prefix.</summary>
+        <returns>
+          <see langword="true" /> if <paramref name="prefix" /> occurs at the start of <paramref name="source" />; otherwise, <see langword="false" />.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This method has greater overhead than other <xref:System.Globalization.CompareInfo.IsPrefix(System.String,System.String,System.Globalization.CompareOptions)> overloads which don't take a `matchLength` argument. Call this overload only if you require the match length information.
```
Comment (gewarren) [inline xml/System.Globalization/CompareInfo.xml:3122]:
```suggestion
This method has greater overhead than other <xref:System.Globalization.CompareInfo.IsPrefix(System.String,System.String,System.Globalization.CompareOptions)> overloads that don't take a `matchLength` argument. Call this overload only if you require the match length information.
```

File: xml/System.Globalization/CompareInfo.xml [hunk 6]
```diff
@@ -3556,13 +3577,24 @@ Use <xref:System.Globalization.CompareInfo.GetSortKeyLength(System.ReadOnlySpan{
...
+        <param name="matchLength">When this method returns, contains the number of characters of <paramref name="source" /> that matched the desired suffix. This may be different than the length of <paramref name="suffix" /> if a linguistic comparison is performed. Set to 0 if the suffix did not match.</param>
+        <summary>Determines whether a string ends with a specific suffix.</summary>
+        <returns>
+          <see langword="true" /> if <paramref name="suffix" /> occurs at the end of <paramref name="source" />; otherwise, <see langword="false" />.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This method has greater overhead than other <xref:System.Globalization.CompareInfo.IsSuffix(System.String,System.String,System.Globalization.CompareOptions)> overloads which don't take a `matchLength` argument. Call this overload only if you require the match length information.
```
Comment (gewarren) [inline xml/System.Globalization/CompareInfo.xml:3592]:
```suggestion
This method has greater overhead than other <xref:System.Globalization.CompareInfo.IsSuffix(System.String,System.String,System.Globalization.CompareOptions)> overloads that don't take a `matchLength` argument. Call this overload only if you require the match length information.
```

File: xml/System.Globalization/CompareInfo.xml [hunk 7]
```diff
@@ -4218,13 +4250,23 @@ Use <xref:System.Globalization.CompareInfo.GetSortKeyLength(System.ReadOnlySpan{
...
+        <param name="options">The <see cref="T:System.Globalization.CompareOptions" /> to use during the search.</param>
+        <param name="matchLength">When this method returns, contains the number of characters of <paramref name="source" /> that matched the desired value. This may be different than the length of <paramref name="value" /> if a linguistic comparison is performed. Set to 0 if <paramref name="value" /> is not found within <paramref name="source" />.</param>
+        <summary>Searches for the last occurrence of a substring within a source string.</summary>
+        <returns>The zero-based index into <paramref name="source" /> where the substring <paramref name="value" /> last appears; or a negative number if <paramref name="value" /> cannot be found within <paramref name="source" />.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This method has greater overhead than other <xref:System.Globalization.CompareInfo.LastIndexOf%2A> overloads which don't take a `matchLength` argument. Call this overload only if you require the match length information.
```
Comment (gewarren) [inline xml/System.Globalization/CompareInfo.xml:4264]:
```suggestion
This method has greater overhead than other <xref:System.Globalization.CompareInfo.LastIndexOf%2A> overloads that don't take a `matchLength` argument. Call this overload only if you require the match length information.
```

File: xml/System/OperatingSystem.xml [hunk 8]
```diff
@@ -302,12 +303,13 @@
...
-        <param name="minor">To be added.</param>
-        <param name="build">To be added.</param>
-        <param name="revision">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="major">The major release number.</param>
+        <param name="minor">The minor release number.</param>
+        <param name="build">The build release number.</param>
+        <param name="revision">The revision release number.</param>
+        <summary>Check for the Android version (returned by <code>uname</code>) with a &gt;= version comparison. This is used to guard APIs that were added in the given Android release.</summary>
```
Comment (gewarren) [inline xml/System/OperatingSystem.xml:310]: Not sure what uname refers to here and in IsFreeBSDVersionAtLeast.

**Pull request merged**